python/lsst/ts/ess/csc/data_client/utils.py

```diff
@@ -0,0 +1,25 @@
+# This file is part of ts_ess_csc.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MAX_ALLOWED_READ_TIMEOUTS"]
+
+# The maximum number of timeouts to allow before raising a TimeoutError.
+MAX_ALLOWED_READ_TIMEOUTS = 5
```

Comment:
I worry that we may want different values for different data clients.

Please consider a config parameter with a sensible default. A slightly less fussy. but less visible, option is a constant in each data client module that has read timeouts.